addons/sms/wizard/sms_composer.py

```diff
@@ -230,7 +230,9 @@ def _action_send_sms_comment_single(self, records=None):
         # on the numbers in the database.
         records = records if records is not None else self._get_records()
         records.ensure_one()
-        if self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number:
+        if not self.number_field_name:
+            self.numbers = self.recipient_single_number_itf or self.recipient_single_number
```

Comment:
Not really clean to update the wizard on the fly before sending it, as it should be consistent at each time. But this require more cleaning, as pointed out by @thiryr , and maybe merge numbers fields to have a unique and coherent computation of recipients. That's outside the scope of this fix so that's ok IMHO.